.github/workflows/local_env_tests.yml

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+name: Local environment tests
+
+on:
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/python/**', 'model/**']
```

Comment:
This looks like a copy-paste error, I don't think we should restrict this to run only when the Python SDK changes. I'm not sure if there's a reasonable set of paths we can restrict this to, we may just want to run it on every PR.

Reply:
Perhaps changes to the following paths should trigger it:
- dev-support/**
- buildSrc/**
- **/build.gradle
- sdks/python/setup.py
- sdks/python/tox.ini

That's almost certainly not exhaustive, but maybe enough.